addons/portal/controllers/portal.py

```diff
@@ -141,6 +141,17 @@ def _get_archive_groups(self, model, domain=None, fields=None, groupby="create_d
         return groups
 
     def _prepare_portal_layout_values(self):
+        """Values for /my & /my/home template rendering.
+
+        Includes the record count for the displayed badges.
+        """
+        return self._prepare_base_portal_values()
+
+    def _prepare_base_portal_values(self):
```

Comment:
Maybe I'm missing something, but I don't get what this does differently.

Reply:
In the portal controllers (other than /my and /my/home), I replace `_prepare_portal_layout_values` by `_prepare_base_portal_values`, to avoid all the `search_count` done for the homepage (the number of records for all the portal sections).
But if a custo is hooked on the `_prepare_portal_layout_values()`, it won't work on the sub-pages of the portal anymore.